spec.bs

```diff
@@ -1295,12 +1317,22 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
       with |biddingSignalsUrl| and true.
     1. If |dataVersion| is not null:
       1. [=map/Set=] |browserSignals|["{{BiddingBrowserSignals/dataVersion}}"] to |dataVersion|.
+    1. Let |fetchSignalDuration| be the [=duration from=] |fetchSignalStartTime| to
+      [=current wall time=], in milliseconds.
+    1. Decrement |perBuyerCumulativeTimeout| by |fetchSignalDuration|.
+    1. If |perBuyerCumulativeTimeout| is negative, then [=iteration/break=];
     1. [=map/For each=] joiningOrigin → |groups| of |perSignalsUrlGenerator|:
       1. [=list/For each=] |ig| of |groups|:
         1. If |ig|'s [=interest group/bidding url=] is null, [=iteration/continue=].
-        1. Let |generatedBid| be the result of [=generate a bid=] given
-          |allTrustedBiddingSignals|, |auctionSignals|, a [=map/clone=] of |browserSignals|,
-          |perBuyerSignals|, |perBuyerTimeout|, |expectedCurrency|, |ig|, and |auctionStartTime|.
+        1. If |perBuyerCumulativeTimeout| is less than |perBuyerTimeout|, then set |perBuyerTimeout|
+          to |perBuyerCumulativeTimeout|.
+        1. Let |generateBidStartTime| be the [=current wall time=].
+        1. Let |generatedBid| be the result of [=generate a bid=] given |allTrustedBiddingSignals|,
+          |auctionSignals|, a [=map/clone=] of |browserSignals|, |perBuyerSignals|,
+          |perBuyerTimeout|, |expectedCurrency|, |ig|, and |auctionStartTime|.
+        1. Let |generateBidDuration| be the [=duration from=] |generateBidStartTime| to
+          [=current wall time=], in milliseconds.

```

Comment:
I think this should be using some sort of monotonic clock rather than wall time?

Reply:
Can this be resolved? Or maybe we're waiting on @morlovich since the last comment ended with a question?

Reply:
checked with @morlovich offline that a timer is monotonic.